### PR TITLE
Yeganeh: task_[249-252]_CosE_STARC

### PR DESCRIPTION
**Pull request by mayank18garg**

### CODE REVIEW

**Comment by swarooprm** [comment]
@yeganehkordi, wondering if you can help review this PR.

**Review by yeganehkordi — COMMENTED** [review]
Thanks for your work!
Some general comments that should be addressed in all the tasks:
- Please update the source fields. We should have one instance per dataset. (i.e., add links and version in the first line)
- It's better to separate the different parts of the input (e.g., question, answer) with a `<sep>` token.
- Make sure that all of the definitions and explanations are grammatically correct. I've mentioned some of the mistakes; please double-check all of the texts.
- Please update the README file with new categories and make the summary of the tasks more concise.

Task 249:
- It's better to split this task into correct answer generation and incorrect answer generation tasks. You can take a look at tasks 004 and 005 in this repo: https://github.com/allenai/natural-instructions-expansion/tree/master/tasks
- I suggest not using a list of options in output because we might have several correct answers for each instance. For example, you can use:
(A)option1 (B)option2 (C)option3
- "Answer generation" or "Incorrect answer generation" is a better category for this task.
-  Maybe replace `the multiple closet` with `multiple closets` and `Answer` with `The answer` in the first example.
- Consider changing `have workers` to `has workers` and `states in sentence` to `states in the sentence`, and correct spelling of the `suits` in the second example.
- Please use an article before `workplace` and `office` in the first negative example.
- Please correct the typo of `arthritis`.

Task 250:
- I think "Text generation" is a better category for this task.
- Maybe split this task into two tasks: one for answering the multiple-choice question and one for generating abstractive explanation. For generating an abstractive explanation, I guess you can drop the choices.
- There are some typos in the explanations; please replace `it's correct answer` with `its correct answer` and `less dense that water` with `less dense than water`, capitalize the first word of the sentences, and use decent articles in the text.

**Comment by swarooprm** [comment]
@mayank18garg, can you address reviewer comments so that I can evaluate?

**Comment by mayank18garg** [comment]
@swarooprm I'm sorry, I'll address all the issues by end of the day, have classes. I hope that's fine!
Thanks!

**Comment by swarooprm** [comment]
> @swarooprm I'm sorry, I'll address all the issues by end of the day, have classes. I hope that's fine!
> Thanks!

Did you address @mayank18garg ? I need to evaluate soon.

**Comment by mayank18garg** [comment]
I have made changes to all the comments by @yeganehkordi. 
Thanks!

**Comment by swarooprm** [comment]
@yeganehkordi, check if you are happy or have any other feedback?

**Comment by yeganehkordi** [comment]
Sorry for the delay! Thanks for the updates! Some minor comments:
Task 249:
- Change the source field to "CosE v1.0 (https://huggingface.co/datasets/cos_e)" in tasks 249, 250, and 252.
- Replace `multiple closet` with `multiple closets` and `one can find multiple closets in house.` with `one can find multiple closets in the house.`.

Task 250:
- Replace `At what place you use computer while gaining knowledge` with `At what place you use the computer while gaining knowledge` and `answer is \"happiness\"` with `the answer is \"happiness\"`(add an article).
- Replace `christmas` with `Christmas`.
- Please change the category to "Text generation".
- Please separate questions and answers in each instance with a `<sep>` token.

Task 251:
- Each element in the output list should be a correct answer. In this task, if we want the users to generate three questions, all of them should be concatenated as one element of the list. I suggest changing the definition to the `generate a question`, in this case, you don't need to modify the instances. We can have many correct answers per instance.
- Change the source to "STARC (https://arxiv.org/pdf/2004.14797.pdf)".
- Replace `You need to generate the three questions from passage` with `You need to generate three questions from the passage` or `You need to generate a question from the passage`. (Based on your decision according to the first comment)
- Replace `All the question` with `All the questions` in the explanations.
- Replace the explanation of the first negative example with `The first question is out of context and could not be answered clearly from the paragraph. While the second question is asking \" why Hannah's job is easy?\", but according to the paragraph her job becomes difficult.`.
- Replace `Second question` with `the Second question` and `out of context and ` with `out of context, and `  in the explanation of the second negative example.

Task 252:
- Change the definition to `You're given a question, three answer options, and a sentence. Predict the correct answer based on the sentence. The sentence is the abstractive explanation based on Commonsense reasoning.` (Some articles and punctuation have been modified).
- Replace the first explanation with `Question is \"In the morning you go to work, and in the evening what?\". The fact clearly states that \" when the day is over, then it's time to leave the work\." Hence, The answer would be leaving work which is the 2nd option among the three choices.` (We need an article before "morning", "evening", and "three", also "would be leaving" seems to be correct here).
- Change the explanation of the first negative example to: `The fact tells us about camping and tent, and the question is where one can find the football. Among the choices, the most appropriate option would be sporting goods store. But since that's not based on the input sentence. Hence, it's not acceptable.`
- Change the second negative explanation to: `The sentence tells us that students do revision and study of material they learned over the course before the exam. The question asked about what one could do before final exams. In the choices, one option is Study, and another option is revision. Both could be the answer, and there is ambiguity in selecting the choices. `
- Please drop the brackets around the choices.

File: tasks/README.md [hunk 1]
```diff
@@ -5,5 +5,7 @@ This directory contains the tasks that are part of this benchmark.
 
 Name | Summary | Category
 ---- | ----------- | --------
-`task1001_squad1.1_question_generation` | Generating guestions (based on SQuAD 1.1) | Question Generation  
-`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
\ No newline at end of file
+`task249_CosE_IncorrectAns_generation` | Writing an IncorrectAns to a question(based on CosE v1.0). | Incorrect Answer Generation
```
Comment (yeganehkordi) [inline tasks/README.md:8]: Task 249: Please change to `Given a sentence and a question, generate an incorrect answer to the question.`
Task 250: Please change to `Given a question and an answer, generate an abstractive explanation for the question based on the answer.`
Task 251: Please change to `Given a paragraph, generate a question from the passage.`
Task 252: Please change to `Given a question, three answer options, and a sentence. Write the correct answer based on the sentence.`

**Comment by swarooprm** [comment]
@mayank18garg, can you address reviewer comments and revise your tasks?

**Comment by mayank18garg** [comment]
I'm on it.

**Comment by yeganehkordi** [comment]
Thanks! Looks good to me! 
Just one point, in task 251, although you don't need to modify the instances, the output of examples shouldn't be a list. So, please choose a single question for each example and update the explanations accordingly.

**Comment by mayank18garg** [comment]
@yeganehkordi Thanks for the update!
So Do I need to remove the two questions from the instance's output?

**Comment by yeganehkordi** [comment]
Yes, it should be something like this:
        {
            "input": "",
            "output": "Where does Clay Cockrell hold his popular therapy sessions?",
            "explanation": " "
        }

**Comment by mayank18garg** [comment]
@yeganehkordi yeah got it, I'll update the positive & negative examples part. But I want to know about the instances. Each instance has three questions, is that format fine?

**Comment by yeganehkordi** [comment]
Yeah, the format of instances is fine.

**Comment by swarooprm** [comment]
Grading completed @mayank18garg

**Pull request closed without merging**